### PR TITLE
Ctor 146 escalation microsoft hyperv 2012 add option for powershell module version in node integration service mode

### LINKED ISSUE #4603: apps::microsoft::hyperv::2012::local::plugin --mode=node-integration-service: critical: integration service 'Heartbeat' status: -/-
Using latest plugins
Using Hyper-V 2019

As you can see we have different VM where integration service reports a problem
image

But we are not able to understand the root cause:
```
PS C:\Windows\system32> Get-VMIntegrationService -VMName SRVSAP11

VMName   Name                    Enabled PrimaryStatusDescription SecondaryStatusDescription
------   ----                    ------- ------------------------ --------------------------
SRVSAP11 Guest Service Interface True    OK
SRVSAP11 Heartbeat               True    OK                       OK
SRVSAP11 Key-Value Pair Exchange True    OK
SRVSAP11 Shutdown                True    OK
SRVSAP11 Time Synchronization    False   OK
SRVSAP11 VSS                     True    OK                       The protocol version of the component installed in...
```
Hearthbeat is enabled and primary and secondary status is OK

```
PS C:\Windows\system32> Get-VM SRVSAP11 | Select-Object *


ParentCheckpointId                  :
ParentCheckpointName                :
VMName                              : SRVSAP11
VMId                                : 1e147df0-542f-48f8-820a-30429dfc9136
CheckpointFileLocation              : C:\ClusterStorage\AlletraDisk3\VMs\SRVSAP11
ConfigurationLocation               : C:\ClusterStorage\AlletraDisk3\VMs\SRVSAP11
SmartPagingFileInUse                : False
SmartPagingFilePath                 : C:\ClusterStorage\AlletraDisk3\VMs\SRVSAP11
SnapshotFileLocation                : C:\ClusterStorage\AlletraDisk3\VMs\SRVSAP11
AutomaticStartAction                : Nothing
AutomaticStartDelay                 : 0
AutomaticStopAction                 : Save
AutomaticCriticalErrorAction        : Pause
AutomaticCriticalErrorActionTimeout : 30
AutomaticCheckpointsEnabled         : False
CPUUsage                            : 1
MemoryAssigned                      : 68719476736
MemoryDemand                        : 70780977152
MemoryStatus                        :
NumaAligned                         : True
NumaNodesCount                      : 1
NumaSocketCount                     : 1
Heartbeat                           : OkApplicationsHealthy
IntegrationServicesState            :
IntegrationServicesVersion          : 0.0
Uptime                              : 46.15:07:11.5860000
OperationalStatus                   : {Ok}
PrimaryOperationalStatus            : Ok
SecondaryOperationalStatus          :
StatusDescriptions                  : {Operating normally}
PrimaryStatusDescription            : Operating normally
SecondaryStatusDescription          :
Status                              : Operating normally
ReplicationHealth                   : NotApplicable
ReplicationMode                     : None
ReplicationState                    : Disabled
ResourceMeteringEnabled             : False
CheckpointType                      : Production
EnhancedSessionTransportType        : VMBus
Groups                              : {}
Version                             : 9.0
VirtualMachineType                  : RealizedVirtualMachine
VirtualMachineSubType               : Generation2
Notes                               :
State                               : Running
ComPort1                            : VMComPort (Name = 'COM 1', VMName = 'SRVSAP11') [Id = 'Microsoft:1E147DF0-542F-48
                                      F8-820A-30429DFC9136\8E3A359F-559A-4B6A-98A9-1690A6100ED7\0', VMId =
                                      '1e147df0-542f-48f8-820a-30429dfc9136']
ComPort2                            : VMComPort (Name = 'COM 2', VMName = 'SRVSAP11') [Id = 'Microsoft:1E147DF0-542F-48
                                      F8-820A-30429DFC9136\8E3A359F-559A-4B6A-98A9-1690A6100ED7\1', VMId =
                                      '1e147df0-542f-48f8-820a-30429dfc9136']
DVDDrives                           : {}
FibreChannelHostBusAdapters         : {}
FloppyDrive                         :
HardDrives                          : {Hard Drive on SCSI controller number 0 at location 0, Hard Drive on SCSI
                                      controller number 1 at location 0, Hard Drive on SCSI controller number 2 at
                                      location 0, Hard Drive on SCSI controller number 3 at location 0}
RemoteFxAdapter                     :
VMIntegrationService                : {Guest Service Interface, Heartbeat, Key-Value Pair Exchange, Shutdown...}
DynamicMemoryEnabled                : False
MemoryMaximum                       : 1099511627776
MemoryMinimum                       : 536870912
MemoryStartup                       : 68719476736
ProcessorCount                      : 8
BatteryPassthroughEnabled           : False
Generation                          : 2
IsClustered                         : True
ParentSnapshotId                    :
ParentSnapshotName                  :
Path                                : C:\ClusterStorage\AlletraDisk3\VMs\SRVSAP11
SizeOfSystemFiles                   : 77824
GuestControlledCacheTypes           : False
LowMemoryMappedIoSpace              : 134217728
HighMemoryMappedIoSpace             : 536870912
LockOnDisconnect                    : Off
CreationTime                        : 13.04.2022 14:03:11
Id                                  : 1e147df0-542f-48f8-820a-30429dfc9136
Name                                : SRVSAP11
NetworkAdapters                     : {Network Adapter}
CimSession                          : CimSession: .
ComputerName                        : SRVHYPV03
IsDeleted                           : False
```

What could be the problem here?
2110

### PR DESCRIPTION
**Pull request by omercier**
# Centreon team

## Description

REFS: CTOR-146

- Modified the powershell script and the perl code to adapt nodeintegrationservices to Hyperv2022
- Added tests with the result of the powershell script converted to JSON 

**Presumably** fixes #4603 (to be confirmed)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Functionality enhancement or optimization (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## How this pull request can be tested ?

The provided tests make the perl processing work with the obtained data without the need to have an actual environment.

## Checklist

- [ ] I have followed the **[coding style guidelines](https://github.com/centreon/centreon-plugins/blob/develop/doc/en/developer/plugins_global.md#5-code-style-guidelines)** provided by Centreon
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (develop).
- [ ] I have implemented automated tests related to my commits.
- [ ] I have reviewed all the help messages in all the .pm files I have modified.
  - [ ] All sentences begin with a capital letter.
  - [ ] All sentences are terminated by a period.
  - [ ] I am able to understand all the help messages, if not, exchange with the PO or TW to rewrite them.

### CODE REVIEW

File: .github/scripts/pod_spell_check.t [hunk 1]
```diff
@@ -5,7 +5,7 @@ use Test::More;
 use Test::Spelling;
 
 if (!@ARGV) {
-    die "Missing perl file to check.";
+    die "Usage: perl pod_spell_check.t module.pm stopwords.t";
```
Comment (Evan-Adam) [inline .github/scripts/pod_spell_check.t:8]: I don't think the stopword file is configurable

File: src/apps/microsoft/hyperv/2012/local/mode/nodeintegrationservice.pm [hunk 2]
```diff
@@ -235,16 +235,34 @@ sub manage_selection {
         my $services = (ref($node->{services}) eq 'ARRAY') ? $node->{services} : [ $node->{services} ];
 
         foreach my $service (@$services) {
+            my $primary_status;
+            my $secondary_status;
+            if ( defined($service->{primary_operational_status}) ) {
```
Comment (Evan-Adam) [inline src/apps/microsoft/hyperv/2012/local/mode/nodeintegrationservice.pm:240]: This if could have been a sub with one parameter and you could use it 2 time

File: tests/resources/spellcheck/stopwords.t [hunk 3]
```diff
@@ -60,7 +64,9 @@ SSH
 Sansymphony
 SureBackup
 TendMicro
```
Comment (Evan-Adam) [inline tests/resources/spellcheck/stopwords.t:66]: You should probably remove this line

File: src/apps/microsoft/hyperv/2012/local/mode/nodeintegrationservice.pm (hunk 2 above)
Comment (omercier) [reply src/apps/microsoft/hyperv/2012/local/mode/nodeintegrationservice.pm:240]: You're right. New commit incoming

Comment (omercier) [reply src/apps/microsoft/hyperv/2012/local/mode/nodeintegrationservice.pm]: Done.

File: tests/resources/spellcheck/stopwords.t (hunk 3 above)
Comment (omercier) [reply tests/resources/spellcheck/stopwords.t]: Done.

**Review by Evan-Adam — APPROVED** [review]

**Review by lucie-dubrunfaut — APPROVED** [review]

**Review by lucie-dubrunfaut — APPROVED** [review]

**Pull request merged**